build-tools/automation/build.groovy

```diff
@@ -40,18 +67,41 @@ timestamps {
         utils = load "${XADir}/build-tools/automation/utils.groovy"
 
         utils.stageWithTimeout('init', 30, 'SECONDS', XADir, true) {    // Typically takes less than a second
+            isCommercial = env.IsCommercial == '1'
+            commercialPath = "external/${env.CommercialDirectory}"
+            packagePath =  "${env.WORKSPACE}/${packageDir}"
+            storageVirtualPath = env.StorageVirtualPath
+
+            skipSigning = env.SkipSigning == '1'
+            skipTest = env.SkipTest == '1'
+
             // Note: PR plugin environment variable settings available here: https://wiki.jenkins.io/display/JENKINS/GitHub+pull+request+builder+plugin
             isPr = env.ghprbActualCommit != null
-            def branch = isPr ? env.GIT_BRANCH : scmVars.GIT_BRANCH
-            def commit = isPr ? env.ghprbActualCommit : scmVars.GIT_COMMIT
+            branch = isPr ? env.GIT_BRANCH : scmVars.GIT_BRANCH
+            commit = isPr ? env.ghprbActualCommit : scmVars.GIT_COMMIT
 
             def buildType = isPr ? 'PR' : 'CI'
 
             echo "Git repo: ${env.GitRepo}"     // Defined as an environment variable in the jenkins build definition
             echo "Job: ${env.JOB_BASE_NAME}"
+            echo "Job name: ${env.JOB_NAME}"
+            echo "Workspace: ${env.WORKSPACE}"
             echo "Branch: ${branch}"
             echo "Commit: ${commit}"
             echo "Build type: ${buildType}"
+            echo "Build number: ${env.BUILD_NUMBER}"
+            echo "IsCommercial: ${isCommercial}"
+
+            if (isCommercial) {
+                echo "Commercial path: ${commercialPath}"
+                storageVirtualPath = "${env.JOB_NAME}-${env.BUILD_NUMBER}/${branch}/${commit}"      // This needs to be unique for commercial builds since the path determines where artifacts.json used for GitHub statuses will be published
+            }
+
+            echo "Package path: ${packagePath}"
+            echo "Storage path: ${storageVirtualPath}"
+
+            echo "SkipSigning: ${skipSigning}"
+            echo "SkipTest: ${skipTest}"
```

Comment:
Why would we want to skip tests?

Reply:
generally we wouldn't, but it serves as a useful policy switch if needed down the road w/o having to update the script itself. initially, i put this in place to expediate testing for just the build & publishing phases for the pipeline script. then decided to expose this in a more official way.